lambdas/data-migration2/src/pdrs.ts

```diff
@@ -0,0 +1,136 @@
+import Logger from '@cumulus/logger';
+import {
+  getRecordCumulusId,
+  PostgresCollectionRecord,
+  PostgresExecutionRecord,
+  PostgresProviderRecord,
+  PostgresPdrRecord,
+  PostgresPdr,
+  tableNames,
+} from '@cumulus/db';
+import { envUtils } from '@cumulus/common';
+import { RecordAlreadyMigrated } from '@cumulus/errors';
+
+import { MigrationSummary } from './types';
+
+const logger = new Logger({ sender: '@cumulus/data-migration/pdrs' });
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+
+/**
+ * Migrate PDR record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordAlreadyMigrated} if record was already migrated
+ */
+export const migratePdrRecord = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<void> => {
+  // Validate record before processing using API model schema
+  Manager.recordIsValid(dynamoRecord, schemas.pdr);
+
+  const existingRecord = await knex<PostgresPdrRecord>(tableNames.pdrs)
+    .where({
+      name: dynamoRecord.pdrName,
+    })
+    .first();
+
+  // Throw error if it was already migrated.
+  if (existingRecord) {
+    throw new RecordAlreadyMigrated(`PDR name ${dynamoRecord.pdrName} was already migrated, skipping.`);
+  }
+
+  const collectionCumulusId = await getRecordCumulusId<PostgresCollectionRecord>(
+    { name: dynamoRecord.collectionId },
+    tableNames.collections,
+    knex
+  );
+
+  const providerCumulusId = await getRecordCumulusId<PostgresProviderRecord>(
+    { name: dynamoRecord.provider },
+    tableNames.providers,
+    knex
+  );
+
+  const executionCumulusId = dynamoRecord.execution
+    ? await getRecordCumulusId<PostgresExecutionRecord>(
+      { arn: dynamoRecord.execution.arn },
+      tableNames.executions,
+      knex
+    )
+    : undefined;
+
+  // Map old record to new schema.
+  const updatedRecord: PostgresPdr = {
+    name: dynamoRecord.pdrName,
+    provider_cumulus_id: providerCumulusId,
+    collection_cumulus_id: collectionCumulusId,
+    execution_cumulus_id: (executionCumulusId === undefined) ? undefined : executionCumulusId,
+    status: dynamoRecord.status,
+    progress: dynamoRecord.progress,
+    pan_sent: dynamoRecord.PANSent,
+    pan_message: dynamoRecord.PANmessage,
+    stats: dynamoRecord.stats,
+    address: dynamoRecord.address,
+    original_url: dynamoRecord.originalUrl,
+    timestamp: dynamoRecord.timestamp ? new Date(dynamoRecord.timestamp) : undefined,
+    duration: dynamoRecord.duration,
+    created_at: new Date(dynamoRecord.createdAt),
+    updated_at: dynamoRecord.updated ? new Date(dynamoRecord.updatedAt) : undefined,
+  };
+
+  await knex(tableNames.pdrs).insert(updatedRecord);
```

Comment:
We should be using the method for creating records on the PDR PG model. Better yet, let's use the `upsert` method, since it already exists and we decided that our data migrations should be doing upserts

Reply:
Definitely agree. Will update